client/src/components/Progress/progress-inner.tsx

```diff
@@ -41,17 +41,24 @@ function ProgressInner({
   const [shownPercent, setShownPercent] = useState(0);
   const [lastShopwnPercent, setLastShownPercent] = useState(0);
   const progressInnerWrap = useRef<HTMLDivElement>(null);
+  const intervalRef = useRef<number | null>(null);
```

Comment:
The changes in this file were because of memory leak warnings when completing a daily challenge.